### ISSUE TITLE
Crash in runtime

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1670]
The below code causes a crash in runtime:
```cs
static List<int> l = new .() ~ delete _;

public static void Main()
{
	IEnumerator<int> e = Environment.TickCount == 0 ? (.)null : l.GetEnumerator();
}
```
Tested with 17dfc96a1fa873b75f8a6fcdac9dc2691f74cfd6

**Comment by disarray2077** [comment]
Also... If you change `Environment.TickCount == 0` to `l != null`, `l` is apparently somehow deleted before its static destructor is called, which causes a double delete error.

**Comment by bfiete** [comment]
Fixed at a23406d9dd242ce8d6b28d771abb4f94f8d567e5
